src/Indicator.vala

```diff
@@ -82,34 +87,40 @@ public class Drop.Indicator : Wingpanel.Indicator {
 	private void setup_notify () {
 		if (notification == null) {
 			Notify.init ("Drop");
-			string summary = "Incoming File";
-			string body = "";
 
-			notification = new Notify.Notification (summary, body, NOTIFICATION_ICON);
+			notification = new Notify.Notification ("", "", NOTIFICATION_ICON);
 		}
 	}
 
 	private void show_notification (string message) {
 		if (!open) {
-			this.notification.update ("Drop share", message, NOTIFICATION_ICON);
+			this.notification.update ("Drop", message, NOTIFICATION_ICON);
 			try {
 				this.notification.show ();
 			} catch (Error E) {}
 		}
 	}
 
-	private void chech_visibility () {
-		if (in_list.transmission_count == 0 && out_list.transmission_count == 0) visible = false;
-		else if (!visible) visible = true;
+	private void check_visibility () {
+		if (in_list.transmission_count == 0 && out_list.transmission_count == 0) {
+		    close ();
```

Comment:
Please use this.close() and this.visible etc. to make more clear where the one who reads this code can find the method.